feat(server) Enable ML support for ARM CPUs

ollywelch opened the pull request:
Hi there, thought I'd create a quick PR to enable the ML image to be built on ARM CPUs. As per [pytorch's website](https://pytorch.org/blog/prototype-features-now-available-apis-for-hardware-accelerated-mobile-and-arm64-builds/), their nightly builds have ARM64 support. Using this would be great for us Raspberry Pi users!

On a side note, love this project, and keen to contribute back. Let me know if there's anything else I need to do for this to be considered :)

ollywelch committed:
Install nightly release of pytorch to enable ML support for arm CPUs

ollywelch committed:
Remove linux/arm/v7 from ML docker builds

alextran1502 commented:
Hello, what is the image size for this change? We use the cpu only image to keep the size small

ollywelch commented:
> Hello, what is the image size for this change? We use the cpu only image to keep the size small

1.6 GB when I built it on my rpi4 - quite beefy

bo0tzz commented:
This is great, thank you! I was going to mention that the armv7 build fails, but I see you already got it :)

Alex: this is still cpu-only as far as I can tell. Olly: AFAIK future gpu support is as easy as swapping `cpu` for `cu118`, but just to be sure, do you know whether the Cuda builds also support arm64?

ollywelch commented:
> 

Not sure about the CUDA builds, would have to check

bo0tzz commented:
It seems like the current machine-learning image is 749MB. @ollywelch is the 1.6GB you mention for an arm-only image, or a multiplatform build?

ollywelch committed:
Add --no-cache-dir to torch installation command in ML image build

ollywelch commented:
> It seems like the current machine-learning image is 749MB. @ollywelch is the 1.6GB you mention for an arm-only image, or a multiplatform build?

@bo0tzz sorry think I quoted the wrong figure there - compressed size is 557MB, just pushed it to [docker hub](https://hub.docker.com/r/ollywelch/immich_machine_learning/tags). Also realised I removed the --no-cache-dir from the first pip install, so I've added that back in now!

ollywelch committed:
Use PIP_NO_CACHE_DIR option in ML build to further decrease image size

ollywelch commented:
Make that 483MB with the latest change :)

bo0tzz reviewed (approved):
I've pulled your image to an arm64 system and it's working flawlessly. Thank you for the PR!
